drivers/tty/serdev/pibridge.c

```diff
@@ -287,55 +290,64 @@ int pibridge_req_gate_tmt(u8 dst, u16 cmd, u8 *snd_buf, u16 snd_len,
 		return -EIO;
 	}
 
-	if (rcv_len != 0) {
-		if (pibridge_recv(rcv_buf, rcv_len) != rcv_len) {
+	crc = pibridge_crc8(0, (u8 *) &pkthdr, sizeof(pkthdr));
+
+	to_receive = min((u16) pkthdr.len, rcv_len);
+	to_discard = pkthdr.len - to_receive;
+
+	if (to_receive) {
+		if (pibridge_recv(rcv_buf, to_receive) != to_receive) {
 			dev_warn_ratelimited(&pibridge_s->serdev->dev,
 				"receive data error in gate-req(len:%d)\n",
-				rcv_len);
+				to_receive);
 			return -EIO;
 		}
 	}
 
-	if (pibridge_recv(&crc_rcv, sizeof(u8)) != sizeof(u8)) {
+	if (to_discard) {
+		/* The provided buffer was too small. Discard the rest of the
+		   received data as well as the following CRC checksum byte. */
+		if (pibridge_discard_timeout(to_discard + 1, tmt))
+			dev_warn_ratelimited(&pibridge_s->serdev->dev,
+				"failed to discard %u bytes within timeout\n",
+				to_discard);
 		dev_warn_ratelimited(&pibridge_s->serdev->dev,
-			"receive crc error in gate-req\n");
-		return -EIO;
+			"received packet truncated (%u bytes missing)\n",
```

Comment:
I was confused at first that there's no `return` statement in the `if (to_discard)` branch, but I guess the CRC check below will fail (with high probability) for such packets, so maybe an explicit `return` indeed isn't necessary.

Reply:
Note that in the discard case the CRC of the data portion is not built, so it wont fail. The caller will wont get a negative return value but only a warning in the log. Should there rather be a negative return value?

Reply:
Right after the `if ... else ...` statement there's a crc check:

```
	if (crc != crc_rcv)
		return -EBADMSG;
```

If the control flow entered the `if (to_discard)` branch, then `crc_rcv` is uninitialized and `crc` contains the CRC calculated over the header. With high probability, that check will fail and as a result, the function is exited with `-EBADMSG`.

An explicit `return` in the `if (to_discard)` branch might be cleaner.

If your intention is to not return an error, you need to change the control flow such that the crc check is skipped in the `if (to_discard)` case.